test/ectest.c

```diff
@@ -1084,6 +1086,86 @@ static int char2_field_tests(void)
     return r;
 }
 
+static int check_bn_mont_ctx(BN_MONT_CTX *mont, BIGNUM *mod, BN_CTX *ctx)
```

Comment:
My opinion is that this test is not really doing much.. 
If the code is wrong then EC related KAT tests will fail.
Other reviewers may disagree.

Reply:
As I learned from a rather embarrassing mistake in the code in an earlier, it doesn't always. That leaves the question of if we care if the KATs pass, but I think we should have this check and if more optimized methods mean this test needs changing down the line, so be it.

Reply:
I think it is useful but it should be moved to ec_internal_test.c